rand_distr/src/normal.rs

```diff
@@ -140,15 +144,34 @@ impl std::error::Error for Error {}
 impl<F> Normal<F>
 where F: Float, StandardNormal: Distribution<F>
 {
-    /// Construct a new `Normal` distribution with the given mean and
-    /// standard deviation.
+    /// Construct, from mean and standard deviation
+    ///
+    /// Parameters:
+    ///
+    /// -   mean (`μ`, unrestricted)
+    /// -   standard deviation (`σ ≥ 0`)
     #[inline]
     pub fn new(mean: F, std_dev: F) -> Result<Normal<F>, Error> {
-        if !(std_dev >= F::zero()) {
-            return Err(Error::StdDevTooSmall);
+        if !std_dev.is_finite() || std_dev < F::zero() {
+            return Err(Error::BadVariance);
         }
         Ok(Normal { mean, std_dev })
     }
+
+    /// Construct, from mean and coefficient of variation
+    ///
+    /// Parameters:
+    ///
+    /// -   mean (`μ`, unrestricted)
+    /// -   coefficient of variation (`cv = σ / μ` where `cv ≥ 0`)
```

Comment:
Pedantically, shouldn't this be `cv = abs(σ / μ)`?

Reply:
You tell me. According to WP it doesn't look like it. I have too limited background in stats myself.

But *if* we do this, doesn't it imply σ may be negative? Currently we prevent that.

Reply:
I was more worried about μ being negative, which would imply that σ must be negative to satisfy cv ≥ 0. That confused me. I don't know how it is formally done in statistics either.

For positive σ, we could also write `cv = σ / abs(μ)`.

Reply:
Looking at Wikipedia, they say that sometimes the definition `cv = σ / abs(μ)` is used, which is equivalent to ours for `σ ≥ 0`. Anyway, I don't think it makes a difference for us, as long as we define it clearly.